assets/src/dashboard/utils/usePagePreviewSize.js

```diff
@@ -110,6 +118,24 @@ export default function usePagePreviewSize(options = {}) {
     availableContainerSpace,
   ]);
 
+  useLayoutEffect(
+    () => setAvailableContainerSpace(getTrueInnerWidth(wpMenuWidth)),
+    [wpMenuWidth]
+  );
+
+  useLayoutEffect(() => {
```

Comment:
This is another case where `useLayoutEffect` doesn't hurt anything, but also doesn't add anything.
Here's the difference in order of events if you were to use this as an effect vs a layout effect:

**Layout Effect**
```
// -> render caused by first run or deps updating
// -> layout effect adds event listener before DOM paints
// -> DOM paints
```

**Effect**
```
// -> render caused by first run or deps updating
// -> DOM paints
// -> Effect adds event listener
```

In either case, the user is never gonna be able to click between a render and a paint. You could think of this being like:
```
const syncAddListener = (listener) => {
  elem.addEventListener('click', listener);
};
syncAddListener(() => console.log('click from sync added Fn'));

// vs
const asyncAddListener = (listener) => {
  setTimeout(() => elem.addEventListener('click', listener), 0);
};
asyncAddListener(() => console.log('click from async added Fn'));
```
with either one, the user's never going to be able to click fast enough for `"click from sync added Fn"` to be logged and not see `"click from async added Fn"` to be logged.